src/DistributedLock.Etcd/EtcdLeaseDistributedLock.cs

```diff
@@ -0,0 +1,198 @@
+                cancellationToken: cancellationTokenSource.Token).ConfigureAwait(false);
+        var actualKey = response.Key;
+        return new EtcdLeaseDistributedLockHandle(actualKey.ToString(), this._etcdClient, leaseId);
+    }
+
+    public string Name => this.LockName;
+
+    public static string GetSafeName(string name)
+    {
+        // TODO figure
+        return DistributedLockHelpers.ToSafeName(name, 1000, s => s);
+    }
+}
+
+/// <summary>
+/// basically similar to AzureBlobLeaseDistributedLock
+/// </summary>
+public class EtcdLeaseOptionsBuilder
+{
+    /// <summary>
+    /// From https://docs.microsoft.com/en-us/rest/api/storageservices/lease-blob:
+    /// "The lock duration can be 15 to 60 seconds, or can be infinite"
+    /// </summary>
+    internal static readonly TimeoutValue MinLeaseDuration = TimeSpan.FromSeconds(15),
+        MaxNonInfiniteLeaseDuration = TimeSpan.FromSeconds(60),
+        DefaultLeaseDuration = TimeSpan.FromSeconds(30);
+
+    private TimeoutValue? _duration, _renewalCadence, _minBusyWaitSleepTime, _maxBusyWaitSleepTime;
+
+    internal EtcdLeaseOptionsBuilder() { }
+
+    /// <summary>
+    /// Specifies how long the lease will last, absent auto-renewal.
+    /// 
+    /// If auto-renewal is enabled (the default), then a shorter duration means more frequent auto-renewal requests,
+    /// while an infinite duration means no auto-renewal requests. Furthermore, if the lease-holding process were to
+    /// exit without explicitly releasing, then duration determines how long other processes would need to wait in 
+    /// order to acquire the lease.
+    /// 
+    /// If auto-renewal is disabled, then duration determines how long the lease will be held.
+    /// 
+    /// Defaults to 30s.
+    /// </summary>
+    public EtcdLeaseOptionsBuilder Duration(TimeSpan duration)
+    {
+        var durationTimeoutValue = new TimeoutValue(duration, nameof(duration));
+        if (durationTimeoutValue.CompareTo(MinLeaseDuration) < 0
+            || (!durationTimeoutValue.IsInfinite && durationTimeoutValue.CompareTo(MaxNonInfiniteLeaseDuration) > 0))
+        {
+            throw new ArgumentOutOfRangeException(nameof(duration), duration,
+                $"Must be infinite or in [{MinLeaseDuration}, {MaxNonInfiniteLeaseDuration}]");
+        }
+
+        this._duration = durationTimeoutValue;
+        return this;
+    }
+
+    /// <summary>
+    /// Determines how frequently the lease will be renewed when held. More frequent renewal means more unnecessary requests
+    /// but also a lower chance of losing the lease due to the process hanging or otherwise failing to get its renewal request in
+    /// before the lease duration expires.
+    /// 
+    /// To disable auto-renewal, specify <see cref="Timeout.InfiniteTimeSpan"/>
+    /// 
+    /// Defaults to 1/3 of the specified lease duration (may be infinite).
+    /// </summary>
+    public EtcdLeaseOptionsBuilder RenewalCadence(TimeSpan renewalCadence)
```

Comment:
Seems like the underlying client doesn't give us control over renewal cadence and that it's always relative to duration. Should we instead just allow renewal to be enabled/disabled?